cmd/fleet/cron.go

```diff
@@ -869,3 +872,97 @@ func cleanupCronStatsOnShutdown(ctx context.Context, ds fleet.Datastore, logger
 		logger.Log("err", "cancel pending cron stats for instance", "details", err)
 	}
 }
+
+func newActivitiesStreamingSchedule(
+	ctx context.Context,
+	instanceID string,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+	auditLogger fleet.JSONLogger,
+) (*schedule.Schedule, error) {
+	const (
+		name     = string(fleet.CronActivitiesStreaming)
+		interval = 5 * time.Minute
+	)
+	logger = kitlog.With(logger, "cron", name)
+	s := schedule.New(
+		ctx, name, instanceID, interval, ds, ds,
+		schedule.WithLogger(logger),
+		schedule.WithJob(
+			"cron_activities_streaming",
+			func(ctx context.Context) error {
+				return cronActivitiesStreaming(ctx, ds, logger, auditLogger)
+			},
+		),
+	)
+	return s, nil
+}
+
+var ActivitiesToStreamBatchCount uint = 500
+
+func cronActivitiesStreaming(
+	ctx context.Context,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+	auditLogger fleet.JSONLogger,
+) error {
```

Comment:
Should we log some stats here to make sure we are streaming the audit logs in, generally, less than 5 mins?

Reply:
Agree. Does this work (already logged by the cron schedule package).
```
level=info ts=2022-12-19T13:50:00.06442Z cron=activities_streaming schedule=activities_streaming status=pending
[...]
level=info ts=2022-12-19T13:50:00.073759Z cron=activities_streaming schedule=activities_streaming status=completed
```